x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotRequest.java

```diff
@@ -0,0 +1,223 @@
+                                          Settings indexSettings, String[] ignoredIndexSettings, boolean waitForCompletion) {
+        this.mountedIndexName = Objects.requireNonNull(mountedIndexName);
+        this.repositoryName = Objects.requireNonNull(repositoryName);
+        this.snapshotName = Objects.requireNonNull(snapshotName);
+        this.snapshotIndexName = Objects.requireNonNull(snapshotIndexName);
+        this.indexSettings = Objects.requireNonNull(indexSettings);
+        this.ignoredIndexSettings = Objects.requireNonNull(ignoredIndexSettings);
+        this.waitForCompletion = waitForCompletion;
+    }
+
+    MountSearchableSnapshotRequest(StreamInput in) throws IOException {
+        super(in);
+        this.mountedIndexName = in.readString();
+        this.repositoryName = in.readString();
+        this.snapshotName = in.readString();
+        this.snapshotIndexName = in.readOptionalString();
+        this.indexSettings = readSettingsFromStream(in);
+        this.ignoredIndexSettings = in.readStringArray();
+        this.waitForCompletion = in.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(mountedIndexName);
+        out.writeString(repositoryName);
+        out.writeString(snapshotName);
+        out.writeOptionalString(snapshotIndexName);
+        writeSettingsToStream(indexSettings, out);
+        out.writeStringArray(ignoredIndexSettings);
+        out.writeBoolean(waitForCompletion);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+       return null;
+    }
+
+    /**
+     * @return the name of the index that will be created
+     */
+    public String mountedIndexName() {
+        return mountedIndexName;
+    }
+
+    /**
+     * @return the name of the repository
+     */
+    public String repositoryName() {
+        return this.repositoryName;
+    }
+
+    /**
+     * @return the name of the snapshot.
+     */
+    public String snapshotName() {
+        return this.snapshotName;
+    }
+
+    /**
+     * @return the name of the index contained in the snapshot
+     */
+    public String snapshotIndexName() {
+        return snapshotIndexName;
+    }
+
+    /**
+     * @return true if the operation will wait for completion
+     */
+    public boolean waitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * @return settings that should be added to the index when it is mounted
+     */
+    public Settings indexSettings() {
+        return this.indexSettings;
+    }
+
+    /**
+     * @return the names of settings that should be removed from the index when it is mounted
+     */
+    public String[] ignoreIndexSettings() {
+        return ignoredIndexSettings;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Generating a `XContent` from a request object is useful for the `RestHighLevelClient` only (we don't really need it for testing the server logic) so I think this code should be moved to the `rest-high-level` project in a different PR. We'll have to upgrade the `RestHighLevelClient` to support other searchable snapshot APIs anyway and we can do it later.

Reply:
Ok, I've removed this in 4a7588f2959f7a43e98061d788e9ff8225441cfa and added a note to #50999 to revisit client support later.